benchmarks/src/main/java/com/slack/kaldb/IndexingBenchmark.java

```diff
@@ -0,0 +1,63 @@
+package com.slack.kaldb;
+
+import com.slack.kaldb.logstore.LogMessage;
+import com.slack.kaldb.logstore.LuceneIndexStoreImpl;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Stream;
+import org.openjdk.jmh.annotations.*;
+
+@State(Scope.Thread)
+public class IndexingBenchmark {
+
+  private final Duration commitInterval = Duration.ofSeconds(5 * 60);
+  private final Duration refreshInterval = Duration.ofSeconds(5 * 60);
+
+  private Path tempDirectory;
+  private MeterRegistry registry;
+  LuceneIndexStoreImpl logStore;
+  private Random random;
+
+  @Setup(Level.Iteration)
+  public void createIndexer() throws IOException {
+    random = new Random();
+    registry = new SimpleMeterRegistry();
+    tempDirectory =
+        Files.createDirectories(
+            Paths.get("jmh", String.valueOf(random.nextInt(Integer.MAX_VALUE))));
+    MeterRegistry registry = new SimpleMeterRegistry();
+    logStore =
+        LuceneIndexStoreImpl.makeLogStore(
+            tempDirectory.toFile(), commitInterval, refreshInterval, registry);
+  }
+
+  @TearDown(Level.Iteration)
+  public void tearDown() throws IOException {
+    logStore.close();
+    try (Stream<Path> walk = Files.walk(tempDirectory)) {
+      walk.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+    }
+    registry.close();
+  }
+
+  @Benchmark
+  public void measureIndexing() {
```

Comment:
just a simple benchmark that adds one document. We will expand on them in the coming PRs